src/webapp/utils/sharing-settings.ts

```diff
@@ -0,0 +1,46 @@
+import i18n from "../../utils/i18n";
+import { Maybe } from "../../types/utils";
+import { NamedRef } from "../../domain/entities/Ref";
+import { Permission } from "../../domain/entities/Permission";
+import { PermissionsDialogProps } from "../components/permissions-dialog/PermissionsDialog";
+
+export function buildSharingDescription(props: Maybe<{ users?: NamedRef[]; userGroups?: NamedRef[] }>) {
```

Comment:
Maybe, users?, userGroups? that's a lot of optional values. Our first instinct for these should be to avoid if possible. I wrote `props: { users: NamedRef[]; userGroups: NamedRef[] }` and it typechecks (other than removing the default set)